src/holon/models/rule_actions/rule_action_add_set.py

```diff
@@ -258,7 +128,9 @@ def add_or_set_items(
                 raise ValueError(f"Value to add cannot be smaller than 0. Given value: {n}")
 
         # get parent type and foreign key field name
-        base_parent_type = filtered_repository.base_model_type  # TODO make sure repository has this
+        base_parent_type = (
```

Comment:
als `filtered_repository` `None` is lijkt het me dat deze functie op regel 121 al een error geeft, en deze case hier dus nooit bereikt kan worden.

Reply:
Klopt, dit was nog oude code zonder scenariorepository. Ik heb het aangepast